pkgs/development/tools/alire/default.nix

```
@@ -0,0 +1,32 @@
+{ stdenv, lib, fetchFromGitHub, gnat, gprbuild }:
+
+stdenv.mkDerivation {
+  pname = "alire";
+  version = "1.3-dev";
+  src = fetchFromGitHub {
+    owner = "alire-project";
+    repo = "alire";
+    rev = "8f44a810";
+    hash = "sha256-kYntEMeVPo1+gRI+Q/6qfSyICXNf8yLT+LcbuyeRu/E=";
+    fetchSubmodules = true;
+  };
+
+  OS = if stdenv.isDarwin then "macOS" else null;
+
+  nativeBuildInputs = [ gnat gprbuild ];
+
+  buildPhase = ''
+    gprbuild -j0 -P alr_env
+  '';
```

Please add `runHook`.

```suggestion
  buildPhase = ''
    runHook preBuild
    
    gprbuild -j0 -P alr_env
    
    runHook postBuild
  '';
```